src/components/NewCommentForm.tsx

```diff
@@ -1,8 +1,87 @@
-import React from 'react';
+import cn from 'classnames';
+import React, { useState } from 'react';
+import { CommentData } from '../types/Comment';
+
+type Props = {
+  postId: number;
+  onSubmit: (data: CommentData & { postId: number }) => void;
+};
+
+export const NewCommentForm: React.FC<Props> = ({ postId, onSubmit }) => {
+  const [formData, setFormData] = useState<CommentData>({
+    name: '',
+    email: '',
+    body: '',
+  });
+
+  const [errors, setErrors] = useState({
+    name: false,
+    email: false,
+    body: false,
+  });
+
+  const [isLoading, setIsLoading] = useState(false);
+
+  const handleChange = (
+    e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>,
+  ) => {
+    const { name, value } = e.target;
+
+    setFormData(prevData => ({
```

Comment:
The `setFormData` and `setErrors` methods are being called multiple times within the same function. This is not recommended as it can lead to unnecessary re-renders and make the code harder to understand. It would be better to combine these into a single `setState` call. You can achieve this by creating a new object with the updated values and then calling `setState` once with this new object.